plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4251,6 +4257,34 @@ public void rebootVM(final Connection conn, final VM vm, final String vmName) th
         }
     }
 
+    protected String skipOrRemoveSR(Connection conn, SR sr) {
```

Comment:
@DaanHoogland With all due respect, I don't see why a protected method can't be documented.

In my point of view, private methods don't have to be documented (but may benefit from it), but protected methods might be treated as public (in terms of documenting). Even Oracle gives an example of a protected constructor with javadoc http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#defaultconstructors. In fact, at Xenserver625StorageProcessor class, this method is seen as "public".

The question here should be: Is this method self-explanatory?
I think that with respect to its act Yes. Althought it could be documented clarifying when this method skips and when it removes the SR.